### PR TITLE
feat(FN-1295): back link and error positioning

### PR DESCRIPTION
**Pull request by JoshBinns2000**
# Introduction

The position of the error summary box and back button on some of the utilisation report upload pages doesn't match the designs.

# Resolution

The back link is placed into the top-level `index.njk` function so that it appears directly below the header. The back link can be used in a specific njk file by setting the `backLink` object via `{% set backLink = { href: '/[path]' } %}`. The back link will not appear if your njk template file doesn't include this line. The resulting appearance of `confirm-and-send.njk` and `check-the-report.njk` are shown below.

The error summary box is moved above the `<h1>` in the relevant njk file.

image
image

### CODE REVIEW

File: portal/templates/index.njk [hunk 1]
```diff
@@ -33,6 +33,17 @@
 
     <div class="govuk-width-container">
 
+      {% from "govuk/components/back-link/macro.njk" import govukBackLink %}
```
Comment (toddym42) [inline portal/templates/index.njk:36]: I think I would expect all imports at the top of the file. Does that work here?

File: portal/templates/index.njk [hunk 2]
```diff
@@ -33,6 +33,17 @@
...
+      {% if backLink and backLink.href %}
+        {{ govukBackLink({
+          text: "Back",
+          href: backLink.href,
+          attributes: {
+              "data-cy": "back-link"
```
Comment (toddym42) [inline portal/templates/index.njk:42]: Indenting is slightly off here. Looks like 2 spaces in this file, though I see this is not consistent across all .njk files.

File: portal/templates/index.njk [hunk 3]
```diff
@@ -33,6 +33,17 @@
 
     <div class="govuk-width-container">
 
+      {% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+      {% if backLink and backLink.href %}
+        {{ govukBackLink({
```
Comment (toddym42) [inline portal/templates/index.njk:38]: Are you able to add a screenshot of how this looks now? Always useful to include a screenshot for visual changes so people can more easily comment on the design as well as the technical implementation.

File: portal/templates/index.njk [hunk 4]
```diff
@@ -33,6 +33,17 @@
 
     <div class="govuk-width-container">
 
+      {% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+      {% if backLink and backLink.href %}
```
Comment (toddym42) [inline portal/templates/index.njk:37]: Should we also make the same change in the `gef-ui` package?

Comment (JoshBinns2000) [reply portal/templates/index.njk:37]: Discussed with @ChristianMcc : we would need to run this change past dtfs and the back link is apparently working in gef-ui, so let's not also change gef-ui

File: portal/templates/index.njk (hunk 1 above)
Comment (JoshBinns2000) [reply portal/templates/index.njk]: I've moved it to the top level of `<script>` but I don't believe I can move it any higher than that

**Review by toddym42 — APPROVED** [review]
Looks great : )

**Review by francescastocco — COMMENTED** [review]
Nice find 👏 in portal the back link is also used on the `user-edit.njk` template, does that fall under 'check with dtfs' @ChristianMcc too?

**Comment by francescastocco** [comment]
> Nice find 👏 in portal the back link is also used on the `user-edit.njk` template, does that fall under 'check with dtfs' too @ChristianMcc?

Oops, I missed that you'd already done this :)

**Review by francescastocco — APPROVED** [review]

**Pull request merged**